enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,1029 @@
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+This enhancement is directly enabling the https://issues.redhat.com/browse/XCMSTRAT-545
+feature for ROSA Hosted Control Planes.
+
+#### Standalone Clusters
+
+This proposal does not have any special implications for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+
+This proposal does not have any special implications for single-node
+deployments or MicroShift.
+
+### Risks and Mitigations
+
+- **Risk**: Using public IngressControllerLB subnets for private clusters.
+  - **Mitigation**: We will reject private cluster installations using public IngressControllerLB subnets (see
+    [Consistent Cluster Scope with IngressControllerLB Subnets Validation](#consistent-cluster-scope-with-ingresscontrollerlb-subnets-validation))
+  - **Mitigation**: For private cluster installations using subnet discovery for IngressControllers, public subnets will be excluded from the 
+    `default` IngressControllers, provided the subnets are correctly tagged.
+- **Risk**: Using public ControlPlaneInternalLB subnets which would expose the internal control plane API load balancer.
+  - **Mitigation**: We will reject using public ControlPlaneInternalLB subnets (see
+    [Consistent Cluster Scope with ControlPlaneLB Subnets Validation](#consistent-cluster-scope-with-controlplanelb-subnets-validation)).
+
+### Drawbacks
+
+- Deprecating `platform.aws.subnets` will be painful for users.
+- Distinguishing between BYO subnets/VPC and installer-created subnets/VPC may be confusing if we need
+  to expand the `platform.aws.vpc.subnets` field in the future to accommodate installer-created
+  subnets configuration.
+  - For example, there may be a future desire to specify the IngressController role for installer-created subnets,
+    and it is unclear how this API design would accommodate that.
+- The all-or-nothing approach in [Automatic vs. Manual Role Selection](#automatic-vs-manual-role-selection) prevents
+  users from specifying only a subset of roles to be automatically determined.
+  - For example, a cluster admin might want to isolate the ClusterNode subnet from the ControlPlaneInternalLB subnet,
+    but leave the IngressControllerLB subnets unspecified to rely on subnet discovery for the `default`
+    IngressController. This combination is not currently supported.
+  - While it may be possible to support mixed automatic and manual role selection in the future, the semantics of
+    that configuration remain unclear (see
+    [Allowing Mixed Automatic and Manual Subnet Selection Alternative](#allowing-mixed-automatic-and-manual-subnet-selection-alternative)).
+
+## Open Questions
+
+- Q: Should the install-config design assume all IngressControllerLB subnets are ClusterNode subnets too?
+  - A: No, we must support dedicated IngressController subnet use cases.
+- Q: When using manual role selection, what subnet does the bootstrap node use? Do we need another role?
```

Comment:
No, we don't need another role. The bootstrap node should be treated as a control plane node.

Reply:
Hmm, I'm not clear on this. Today for `platform.aws.subnets`, it will [use](https://github.com/openshift/installer/blob/fc463579baf4ecd75efbcc0aa2610a71f8186f95/pkg/asset/machines/clusterapi.go#L107) all public subnets if publish=external, or private if publish=internal. Since ClusterNode subnets are usually private subnets, I don't think it will be able to use them. 

With the new API, I suppose this is easy when roles are not specified, as it can easily do the same thing. But when roles are specified, my question is: should it ignore the specified roles and use all public or all private subnets in `platform.aws.vpc.subnets`?

I agree it's probably an implementation detail rather than something we should expose as a new role, I'll document it as such.

Reply:
I didn’t consider that the boostrap node on public clusters has a public ip and therefore needs to go onto a public subnet. That is a good point and worth ironing out.

So I take back that we don’t need another role: perhaps that is the best ux.

Reply:
Hmm, yea the more I think about it, the more it feels like another assumption the installer would need to make. Making it into a role would remove the installer's assumption for manual role selection mode. 

And to be clear, automatic role selection mode is going to make assumptions, but I that's it's goal, as it's the opinionated default. I think we need to keep manual mode as assumption-free as possible. So let me know if you think of any more "roles" of subnets.